From 6e26bf31fed7321baca58a87691ab3310afc9840 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Mon, 3 Dec 2012 12:16:56 +0100
Subject: [PATCH] Do not recommend how to configure DNS in error message

The best way to configure DNS depends on the environment and no general
recommendations should be given by the CLI or Web UI. Especially
forwarders should not be recommended by only be option of last resort.

Fixes https://fedorahosted.org/freeipa/ticket/3261
---
 ipalib/plugins/trust.py | 22 +++++++++-------------
 1 file changed, 9 insertions(+), 13 deletions(-)

diff --git a/ipalib/plugins/trust.py b/ipalib/plugins/trust.py
index b4a3d7697c620c38144ebaf459a6b3b97506f8ce..2019d910b18ea507b9d05f5b6165e7b6d9a43e4e 100644
--- a/ipalib/plugins/trust.py
+++ b/ipalib/plugins/trust.py
@@ -354,19 +354,15 @@ sides.
                             instructions.append(_("Forward policy is defined for it in IPA DNS, "
                                                    "perhaps forwarder points to incorrect host?"))
                     except (errors.NotFound, KeyError) as e:
-                        instructions.append(_("IPA manages DNS, please configure forwarder to "
-                                               "'%(domain)s' domain using following CLI command. "
-                                               "Make sure to replace DNS_SERVER and IP_ADDRESS by "
-                                               "actual values corresponding to the trusted domain's "
-                                               "DNS server:") % dict(domain=keys[-1]))
-                        # tab character at the beginning of a multiline error message will be replaced
-                        # in the web UI by a colorful hint. Does not affect CLI.
-                        instructions.append(_("\tipa dnszone-add %(domain)s --name-server=[DNS_SERVER] "
-                                               "--admin-email='hostmaster@%(domain)s' "
-                                               "--force --forwarder=[IP_ADDRESS] "
-                                               "--forward-policy=only") % dict(domain=keys[-1]))
-                        instructions.append(_("When using Web UI, please create DNS zone for domain '%(domain)s' "
-                                               "first and then set forwarder and forward policy.") % dict(domain=keys[-1]))
+                        instructions.append(_("IPA manages DNS, please verify "
+                                              "your DNS configuration and "
+                                              "make sure that service records "
+                                              "of the '%(domain)s' domain can "
+                                              "be resolved. Examples how to "
+                                              "configure DNS with CLI commands "
+                                              "or the Web UI can be found in "
+                                              "the documentation. " ) %
+                                              dict(domain=keys[-1]))
                 else:
                     instructions.append(_("Since IPA does not manage DNS records, ensure DNS "
                                            "is configured to resolve '%(domain)s' domain from "
-- 
1.7.11.7

